### ISSUE TITLE
Error in phpcd, caused by return self in the interface

### ISSUE THREAD

**Issue by Xerkus** [phpvim/phpcd.vim#27]
Stumbled upon error thrown in phpcd: 
`"Uncaught ReflectionException: Class Psr\Http\Message\defines does not exist in .../phpcd.vim/php/PHPCD.php:328`

```
Stack trace:                                                                                                                                                      
#0 .../phpcd.vim/php/PHPCD.php(328): ReflectionClass->__construct('Psr\\Http\\Messag...')                                            
#1 .../phpcd.vim/php/PHPCD.php(79): PHPCD->classInfo('Psr\\Http\\Messag...', 'has', false, 1)                                        
#2 .../phpcd.vim/php/RpcServer.php(120): PHPCD->info('Psr\\Http\\Messag...', 'has', false, 1)                                        
#3 .../phpcd.vim/php/RpcServer.php(94): RpcServer->doRequest(Array, Array)                                                           
#4 .../phpcd.vim/php/RpcServer.php(74): RpcServer->onRequest(Array)                                                                  
#5 .../phpcd.vim/php/RpcServer.php(46): RpcServer->onMessage(Array)
```

Obviously it is not where it originates.

Dependency and code to reproduce:
`composer require psr/http-message ~1.0`

```
<?php

use Psr\Http\Message\RequestInterface;

class Test
{
    public function test(RequestInterface $request)
    {
        $request->hasHeader('Test');
        $request = $request->withoutHeader('Test');

        $request->

    }
} 
```

Note the assignment. `withoutHeader()` method is defined at https://github.com/php-fig/http-message/blob/master/src/MessageInterface.php#L164 and inherited by `RequestInterface`. Docblock for the method have `@return self` and that triggers error condition

**Comment by lvht** [comment]
Fixed at fd6fafc4e043233eec91c7db943b5a56d044502b . Thank you @Xerkus so much. :+1:

**Comment by Xerkus** [comment]
Getting another error, same stacktrace but `class Array does not exist` this time, on Prophecy mock.
Case sensitive primitive types detection?
Also, I think exception should be caught and handled to prevent error.
